server/src/main/java/org/opensearch/gateway/PrimaryShardAllocator.java

```diff
@@ -457,22 +501,32 @@ private static NodesToAllocate buildNodesToAllocate(
 
     protected abstract FetchResult<NodeGatewayStartedShards> fetchData(ShardRouting shard, RoutingAllocation allocation);
 
-    private static class NodeShardsResult {
-        final List<NodeGatewayStartedShards> orderedAllocationCandidates;
+    /**
+     * This class encapsulates the result of a call to {@link #buildNodeShardsResult}
+     */
+    protected static class NodeShardsResult {
+        final NodeShardStates orderedAllocationCandidates;
         final int allocationsFound;
 
-        NodeShardsResult(List<NodeGatewayStartedShards> orderedAllocationCandidates, int allocationsFound) {
+        NodeShardsResult(NodeShardStates orderedAllocationCandidates, int allocationsFound) {
             this.orderedAllocationCandidates = orderedAllocationCandidates;
             this.allocationsFound = allocationsFound;
         }
     }
 
-    static class NodesToAllocate {
-        final List<DecidedNode> yesNodeShards;
-        final List<DecidedNode> throttleNodeShards;
-        final List<DecidedNode> noNodeShards;
-
-        NodesToAllocate(List<DecidedNode> yesNodeShards, List<DecidedNode> throttleNodeShards, List<DecidedNode> noNodeShards) {
+    /**
+     * This class encapsulates the result of a call to {@link #buildNodesToAllocate(RoutingAllocation, NodeShardStates, ShardRouting, boolean)}
+     * */
+    protected static class NodesToAllocate {
```

Comment:
We can avoid changes which are not required. Also, check if we need a protected access for this class.

Reply:
I have reverted this code change now